doc/releases/migration-guide-3.6.rst

```diff
@@ -56,6 +56,34 @@ enable all optional modules, and then run ``west update`` again.
 Device Drivers and Device Tree
 ==============================
 
+* The :dtcompatible:`nxp,pcf8574` driver has been renamed to
+  :dtcompatible:`nxp,pcf857x`. (:github:`67054`) to support pcf8574 and pcf8575.
+  The Kconfig option has been renamed from ``CONFIG_GPIO_PCF8574`` to
+  ``CONFIG_GPIO_PCF857X``. The Device Tree can be configured as follows:
+
+  .. code-block:: devicetree
+
+    &i2c {
+      status = "okay";
+      pcf8574: pcf857x@20 {
+          compatible = "nxp,pcf8574";
```

Comment:
How is it supposed to work if you had removed this compatible? IMO. it should be replaced with `pcf857x` instead.